tests/acceptance/00_basics/04_bundles/agent-bundle-class-bundle-scope-default-perspective.cf

```diff
@@ -0,0 +1,73 @@
+#!/var/cfengine/bin/cf-agent -f-
+body common control
+{
+      inputs => { "../../default.cf.sub" };
+      bundlesequence => {
+                          default("$(this.promise_filename)") };
+      version => "1.0";
+}
+body file control {  namespace => "test"; }
+bundle agent example
+{
+  classes:
+    "TRIGGERED"
+      expression => "any",
+      scope => "bundle"; # Bundle scoped classes are NOT visible from other bundles
+}
+body file control {  namespace => "default"; }
```

Comment:
What is this double `body file control` supposed to do? And what makes you think it should be like that?

Reply:
The first one set the namespace to test, the second one returned the namespace to default. So `bundle agent example` is in the `test` namespace but `bundle agent test` is in the default namespace.

Reply:
> Please fix the typos. Also, is there a ticket for this issue?

ACK, and yes, ENT-4681 is the related ticket

Reply:
I am quite surprised that the double `body file control` just works like this, sequentially. That looks almost like a crime! :wink:

Reply:
I agree with @vpodzime this looks insane to me. The ordering of bodies and bundles within a file shouldn't matter, at all. Also a `body file control` is the control body for the file no? It should be for the whole file. I would expect the last body to overwrite the attributes of the first one, or an error.

Reply:
Yeah, but this is how it was designed.

https://docs.cfengine.com/docs/3.12/reference-components-file_control_promises.html

```
This directive can be given multiple times within any file, outside of body and bundle definitions.
```